_studio/mfx_lib/shared/src/mfx_mpeg2_encode_vaapi.cpp

```diff
@@ -380,18 +387,74 @@ mfxStatus VAAPIEncoder::QueryEncodeCaps(ENCODE_CAPS & caps)
     caps.VCMBitrateControl = 0; // Video conference mode
     caps.HeaderInsertion   = 0; // We will provide headers (SPS, PPS) in binary format to the driver
     caps.MbQpDataSupport   = 1;
-    caps.SkipFrame         = 1;
 
     caps.SliceIPBOnly      = 1;
-    caps.MaxNum_Reference  = 1;
-    caps.MaxPicWidth       = 1920;
-    caps.MaxPicHeight      = 1088;
     caps.NoInterlacedField = 0; // Enable interlaced encoding
-    caps.SliceStructure    = 1; // 1 - SliceDividerSnb; 2 - SliceDividerHsw; 3 - SliceDividerBluRay; the other - SliceDividerOneSlice
 
+    std::map<VAConfigAttribType, size_t> idx_map;
```

Comment:
I see we use this scheme of vector + map to handle caps in several places... I think we need a dedicated class which can handle this and we can put it in mfx_utils.h

Reply:
Out of scope of this PR. If you @mgonchar have a BW to implement or (even better) to work with Alina to guide her to implement - please, do, but in separate PR which will cover other places in the code.